File: BackEnd/src/main/java/com/tienda_ropa/ecommerce/model/Pedido.java
package com.tienda_ropa.ecommerce.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.tienda_ropa.ecommerce.model.enums.Estado;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.*;

@Entity
@Table(name = "pedido")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Pedido extends Master {

    private LocalDate fecha;

    @Enumerated(EnumType.STRING)
    private Estado estado;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    @JsonBackReference("cliente-pedido")
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "domicilio_id", nullable = false)
    private Domicilio domicilio;

    @OneToMany(mappedBy = "pedido", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference("pedido-detalle")
    private Set<PedidoDetalle> detalles = new HashSet<>();
}

File: FrontEnd/src/models/Pedido.ts
import type Cliente from "./Cliente";
import type PedidoDetalle from "./PedidoDetalle";
import type Domicilio from "./Domicilio.ts";
import type { Estado } from "./enums/Estado.ts";

export default class Pedido {
    id?: number;
    fecha: Date = new Date();
    estado!: Estado;
    cliente!: Cliente;
    detalles: PedidoDetalle[] = [];
    domicilio!: Domicilio;
    eliminado!: boolean;
}